sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -618,6 +618,9 @@ object CombineLimits extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case ll @ Limit(le, nl @ Limit(ne, grandChild)) =>
       Limit(If(LessThan(ne, le), ne, le), grandChild)
+
+    case s @ Sort(_, _, Limit(exp, grandChild)) =>
```

Comment:
maybe add some explanation saying it is more efficient because it is expensive to do total ordering in distributed setting.